frontend/public/components/resource-dropdown.tsx

```diff
@@ -90,63 +64,82 @@ const ResourceListDropdown_: React.SFC<ResourceListDropdownProps> = (props) => {
-  const items = resources.map((model) => (
-    <DropdownItem
+
+  // Create select option for each resource.
+  const items = resources.toArray().map((model) => (
+    <SelectOption
       key={referenceForModel(model)}
-      model={model}
-      showGroup={isDup(model.kind)}
-      checked={isKindSelected(referenceForModel(model))}
-    />
-  )) as OrderedMap<string, JSX.Element>;
+      value={model}
+      isChecked={isKindSelected(referenceForModel(model))}
+    >
+      <span className={'co-resource-item'}>
+        <span className="co-resource-icon--fixed-width">
+          <ResourceIcon kind={referenceForModel(model)} />
+        </span>
+        <span className="co-resource-item__resource-name">
+          <span>
+            {model.kind}
+            {model.badge && (
+              <span className="co-resource-item__tech-dev-preview">{model.badge}</span>
+            )}
+          </span>
+          {isDup(model.kind) && (
+            <span className="co-resource-item__resource-api text-muted co-truncate show co-nowrap small">
+              {model.apiGroup || 'core'}/{model.apiVersion}
+            </span>
+          )}
+        </span>
+      </span>
+    </SelectOption>
+  ));
+
   // Add an "All" item to the top if `showAll`.
-  const allItems = (showAll
-    ? OrderedMap({
-        All: (
-          <>
-            <span className="co-resource-item">
-              <Checkbox id="all-resources" isChecked={isKindSelected('All')} />
-              <span className="co-resource-icon--fixed-width">
-                <ResourceIcon kind="All" />
-              </span>
-              <span className="co-resource-item__resource-name">All Resources</span>
+  const allItems = showAll
+    ? [
+        <SelectOption key="All" value="All" isChecked={isKindSelected('All')}>
+          <span className={'co-resource-item'}>
+            <span className="co-resource-icon--fixed-width">
+              <ResourceIcon kind="All" />
             </span>
-          </>
-        ),
-      }).concat(items)
-    : items
-  ).toJS() as { [s: string]: JSX.Element };
-
-  const autocompleteFilter = (text, item) => {
-    const { model } = item.props;
-    if (!model) {
-      return false;
+            <span className="co-resource-item__resource-name">All Resources</span>
+          </span>
+        </SelectOption>,
+      ].concat(items)
+    : items;
+
+  const onFilter = (event) => {
+    const textInput = event.target.value;
+    if (textInput === '') {
+      return allItems;
     }
+    return allItems.filter((item) => fuzzy(_.toLower(textInput), _.toLower(item.props.value.kind)));
+  };
 
-    return fuzzy(_.toLower(text), _.toLower(model.kind));
+  const onToggle = (isOpen) => {
+    setExpanded(isOpen);
   };
 
-  const handleSelected = (value: string) => {
-    onChange(referenceForModel(modelFor(value)));
+  const handleSelected = (event, value: K8sKind) => {
+    value === 'All' ? onChange(value) : onChange(referenceForModel(value));
```

Comment:
Probably not new, but does this break if we have a resource with kind value `All`?

Reply:
This fixes a previous bug (also fixed on master now) where it would break and not let you select resources with kind value "All" used on the Events page. Is there another resource that uses the same kind?